python/lsst/pipe/tasks/processCcdWithFakes.py

```diff
@@ -343,29 +359,38 @@ def runQuantum(self, butlerQC, inputRefs, outputRefs):
             expId, expBits = butlerQC.quantum.dataId.pack("visit_detector", returnMaxBits=True)
             inputs['exposureIdInfo'] = ExposureIdInfo(expId, expBits)
 
+        expWcs = inputs["exposure"].getWcs()
+        tractId = inputs["skyMap"].findTract(
+            expWcs.pixelToSky(inputs["exposure"].getBBox().getCenter())).tract_id
         if not self.config.doApplyExternalGlobalSkyWcs and not self.config.doApplyExternalTractSkyWcs:
-            inputs["wcs"] = inputs["exposure"].getWcs()
-
+            inputs["wcs"] = expWcs
         elif self.config.doApplyExternalGlobalSkyWcs:
             externalSkyWcsCatalog = inputs["externalSkyWcsGlobalCatalog"]
             row = externalSkyWcsCatalog.find(detectorId)
             inputs["wcs"] = row.getWcs()
-
         elif self.config.doApplyExternalTractSkyWcs:
-            externalSkyWcsCatalog = inputs["externalSkyWcsTractCatalog"]
+            externalSkyWcsCatalogList = inputs["externalSkyWcsTractCatalog"]
+            for externalSkyWcsCatalogRef in externalSkyWcsCatalogList:
+                if externalSkyWcsCatalogRef.dataId["tract"] == tractId:
+                    externalSkyWcsCatalog = externalSkyWcsCatalogRef.get(
+                        datasetType=self.config.connections.externalSkyWcsTractCatalog)
+                    break
             row = externalSkyWcsCatalog.find(detectorId)
             inputs["wcs"] = row.getWcs()
 
         if not self.config.doApplyExternalGlobalPhotoCalib and not self.config.doApplyExternalTractPhotoCalib:
             inputs["photoCalib"] = inputs["exposure"].getPhotoCalib()
-
         elif self.config.doApplyExternalGlobalPhotoCalib:
             externalPhotoCalibCatalog = inputs["externalPhotoCalibGlobalCatalog"]
             row = externalPhotoCalibCatalog.find(detectorId)
             inputs["photoCalib"] = row.getPhotoCalib()
-
         elif self.config.doApplyExternalTractPhotoCalib:
-            externalPhotoCalibCatalog = inputs["externalPhotoCalibTractCatalog"]
+            externalPhotoCalibCatalogList = inputs["externalPhotoCalibTractCatalog"]
+            for externalPhotoCalibCatalogRef in externalPhotoCalibCatalogList:
+                if externalPhotoCalibCatalogRef.dataId["tract"] == tractId:
+                    externalPhotoCalibCatalog = externalPhotoCalibCatalogRef.get(
+                        datasetType=self.config.connections.externalSkyWcsTractCatalog)
```

Comment:
@morriscb Shouldn't this be the externalPhotoCalibCatalog connection instead of the SkyWcsTractCatalog? @sr525 spotted this.

Reply:
+1 again - seems like a copy/paste typo.